docs/reference/api.md

```diff
@@ -210,3 +210,57 @@ This is an internal endpoint used by `slicer vm exec` to obtain a shell. It requ
 
 `/vm/{hostname}/exec`
 
+## Manage secrets
+
+HTTP GET
+
+`/secrets`
+
+```json
+[
+  {
+    "name": "my-secret",
+    "size": 1024,
+    "permissions": "0600",
+    "uid": 1000,
+    "gid": 1000
+  }
+]
+```
+
+HTTP POST
+
+`/secrets`
+
+Create a secret with base64-encoded data:
+
+```json
+{
+  "name": "my-secret",
+  "data": "base64-encoded-content",
```

Comment:
I remember K8s supporting base64 or literal/binary values.

I can't remember what we did in openfaas in the end..